exercises/practice/run-length-encoding/RunLengthEncodingTest.php

```diff
@@ -31,59 +9,154 @@ public static function setUpBeforeClass(): void
+            encode('  hsqq qww  ')
+        );
     }
 
-    public function testDecodeSimple(): void
+    /**
+     * uuid: abf176e2-3fbd-40ad-bb2f-2dd6d4df721a
+     * @testdox Run-length encode a string - lowercase characters
+     */
+    public function testEncodeLowercaseCharacters(): void
     {
-        $this->assertEquals('AABBBCCCC', decode('2A3B4C'));
+        $this->assertEquals(
+            '2a3b4c',
+            encode('aabbbcccc')
+        );
     }
 
-    public function testEncodeWithSingleValues(): void
+    /**
+     * uuid: 7ec5c390-f03c-4acf-ac29-5f65861cdeb5
+     * @testdox Run-length decode a string - empty string
+     */
+    public function testDecodeEmptyString(): void
     {
         $this->assertEquals(
-            '12WB12W3B24WB',
-            encode('WWWWWWWWWWWWBWWWWWWWWWWWWBBBWWWWWWWWWWWWWWWWWWWWWWWWB')
+            '',
+            decode('')
         );
     }
 
-    public function testDecodeWithSingleValues(): void
+    /**
+     * uuid: ad23f455-1ac2-4b0e-87d0-b85b10696098
+     * @testdox Run-length decode a string - single characters only
+     */
+    public function testDecodeSingleCharactersOnly(): void
+    {
+        $this->assertEquals(
+            'XYZ',
+            decode('XYZ')
+        );
+    }
+    /**
+     * uuid: 21e37583-5a20-4a0e-826c-3dee2c375f54
+     * @testdox Run-length decode a string - string with no single characters
+     */
+    public function testDecodeStringWithNoSingleCharacters(): void
+    {
+        $this->assertEquals(
+            'AABBBCCCC',
+            decode('2A3B4C')
+        );
+    }
+    /**
+     * uuid: 1389ad09-c3a8-4813-9324-99363fba429c
+     * @testdox Run-length decode a string - single characters with repeated characters
+     */
+    public function testDecodeSingleCharactersWithRepeatedCharacters(): void
     {
         $this->assertEquals(
             'WWWWWWWWWWWWBWWWWWWWWWWWWBBBWWWWWWWWWWWWWWWWWWWWWWWWB',
             decode('12WB12W3B24WB')
         );
     }
-
+    /**
+     * uuid: 3f8e3c51-6aca-4670-b86c-a213bf4706b0
+     * @testdox Run-length decode a string - multiple whitespace mixed in string
+     */
     public function testDecodeMultipleWhitespaceMixedInString(): void
     {
-        $this->assertEquals('  hsqq qww  ', decode('2 hs2q q2w2 '));
+        $this->assertEquals(
+            '  hsqq qww  ',
+            decode('2 hs2q q2w2 ')
+        );
     }
-
-    public function testEncodeDecodeCombination(): void
+    /**
+     * uuid: 29f721de-9aad-435f-ba37-7662df4fb551
+     * @testdox Run-length decode a string - lowercase string
+     */
+    public function testDecodeLowercaseString(): void
     {
-        $this->assertEquals('zzz ZZ  zZ', decode(encode('zzz ZZ  zZ')));
+        $this->assertEquals(
+            'aabbbcccc',
+            decode('2a3b4c')
+        );
+    }
+    /**
+     * uuid: 2a762efd-8695-4e04-b0d6-9736899fbc16
+     * @testdox Run-length decode a string - encode followed by decode gives original string
```

Comment:
Once again too much copy&paste... this is not part of one of the other `cases`:

```suggestion
     * @testdox encode followed by decode gives original string
```